.github/workflows/nightly-release.yaml

```diff
@@ -0,0 +1,43 @@
+name: Nightly Release
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: "0 2 * * *" # release nightly at 2AM
+
+permissions:
+  contents: read
+
+env:
+  # renovate: datasource=github-tags depName=goreleaser-pro packageName=goreleaser/goreleaser-pro
+  GORELEASER_PRO_VERSION: v2.7.0
+
+jobs:
+  nightly-release:
+    name: Nightly Release
+    runs-on: ubuntu-latest
+
+    permissions:
+      contents: write
+
+    steps:
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        with:
+          fetch-depth: 0
+          token: ${{ secrets.GH_PAT }}
```

Comment:
In the base release workflow this secret has been renamed from `GH_PAT` to `GITHUB_TOKEN`. Why was it renamed there and here we used the old name?

Reply:
But I think for extra safety in this PR, i'll revert the change in the base release pipeline or put it into a separate PR

Reply:
i separated the change out into https://github.com/open-telemetry/opentelemetry-collector-releases/pull/1043